### PR TITLE
feat: add option to hide submit button in the confirmation template

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**
This PR aims to add an option to hide the submit button in the confirmation template.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23085?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2129

## **Manual testing steps**
Note: It won't be possible to test this option as it was not adopted by any confirmation template.
I recorded a video to show the behaviour with the new option while I was testing.

Test regression adding chain and snaps both are using confirmation template.
Chain:
1. Go to https://chainlist.org/ page
2. Add a chain to MetaMask
3. Successfully added

Snap:
1. Go to https://snaps.metamask.io/explore/ page
2. Install a snap to MetaMask
3. Successfully installed

## **Screenshots/Recordings**

[hide-submit-btn.webm](https://github.com/MetaMask/metamask-extension/assets/45455812/bb9eebd1-4440-4a19-9ee5-a77160147d21)

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/confirmations/confirmation/confirmation.js [hunk 1]
```diff
@@ -193,6 +193,7 @@ function useTemplateState(pendingConfirmation) {
 
 export default function ConfirmationPage({
   redirectToHomeOnZeroConfirmations = true,
+  hideSubmitButton = false,
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/confirmation/confirmation.js:196]: Does this need to be a component level property?

We'll only have one instance for the client so is it more explicit to always make this an approval template decision?

File: ui/pages/confirmations/confirmation/confirmation.js [hunk 2]
```diff
@@ -541,11 +542,13 @@ export default function ConfirmationPage({
             <MetaMaskTemplateRenderer sections={alert.content} />
           </Callout>
         ))}
+        hideSubmitButton={templatedValues.hideSubmitButton ?? hideSubmitButton}
```
[the remark this answers is not in the document]
Comment (dan437) [reply ui/pages/confirmations/confirmation/confirmation.js:545]: That would be great if the submit button would be hidden if we don't pass the `onSubmit` prop.

[the remark this answers is not in the document]
Comment (dan437) [reply ui/pages/confirmations/confirmation/confirmation.js:545]: When I don't pass `onSubmit`, I see that it still renders the `confirmation-footer__actions` section, which has a `border-top` style. I would say if we don't want to display any default buttons, the whole `confirmation-footer` section should not be rendered.

image

File: ui/pages/confirmations/confirmation/confirmation.js (hunk 1 above)
Comment (dbrans) [reply ui/pages/confirmations/confirmation/confirmation.js]: Done.

File: ui/pages/confirmations/confirmation/confirmation.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (dbrans) [reply ui/pages/confirmations/confirmation/confirmation.js]: Done.

**Comment by dbrans** [comment]
Here's what it looks like if you do not render the actions in the confirmation footer (contrived example).

image

Specifically the alerts are still visible.

**Comment by dbrans** [comment]
Linting will be fixed by https://github.com/MetaMask/metamask-extension/pull/23204

**Review by matthewwalsh0 — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Comment by dbrans** [comment]
# Manual verification 
Adding BNB Chain on [chainlist.org](chainlist.org)

## Before
image

## After
Manually setting hideSubmitButton = true in template [add-ethereum-chain.js](https://github.com/MetaMask/metamask-extension/blob/3a6cd48871b0bb5f14e6fa422922a701edba34e2/ui/pages/confirmations/confirmation/templates/add-ethereum-chain.js#L199)

image

**Pull request merged**